src/Item.php

```diff
@@ -2349,6 +2351,81 @@ public function get_source()
 
         return null;
     }
+
+    /**
+     * @uses idna_convert If available, this will convert an IDN in $url
+     * @see https://github.com/algo26-matthias/idna-convert
+     */
+    private function create_enclosure(
```

Comment:
Why move this out? It seems to me like it goes the opposite way from moving the IDN handling into the HTTP request component.

Reply:
I moved it here, because this is the only place where `Enclosure`s are created and `Item` already implements `RegistryAware`. If we want the idna_convert inside `Enclosure` we would have to implement `RegistryAware` there too.